Automattic/wp-calypso
# Purchases: Convert ManagePurchase to TypeScript

## Proposed Changes

This PR converts the `ManagePurchase` component to TypeScript while making as few functional changes as possible.

Screenshot of the component with devtools showing its location on the page (it's most of the page):

<img width="741" alt="Screenshot 2023-05-08 at 5 56 20 PM" src="https://user-images.githubusercontent.com/2036909/236946867-d4663c73-74b6-4500-8807-e58c324ab170.png">

## Testing Instructions

- Visit `/me/purchases/` for an account with at least one active purchase.
- Click through to a purchase in the list.
- Verify that the purchase page looks the same before and after this PR.
